pkg/datarepair/checker/checker.go

```diff
@@ -146,3 +160,67 @@ func (c *checker) offlineNodes(ctx context.Context, nodeIDs storj.NodeIDList) (o
 	}
 	return offline, nil
 }
+
+// Find invalidNodes by checking the audit results that are place in statdb
+func (c *checker) invalidNodes(ctx context.Context, nodeIDs storj.NodeIDList) (invalidNodes []int32, err error) {
+	// filter if nodeIDs have invalid pieces from auditing results
+	findValidNodesReq := &sdbpb.FindValidNodesRequest{
+		NodeIds: nodeIDs,
+		MinStats: &pb.NodeStats{
+			AuditSuccessRatio: 0, // TODO: update when we have stats added to statdb
+			UptimeRatio:       0, // TODO: update when we have stats added to statdb
+			AuditCount:        0, // TODO: update when we have stats added to statdb
+		},
+	}
+
+	resp, err := c.statdb.FindValidNodes(ctx, findValidNodesReq)
+	if err != nil {
+		return nil, Error.New("error getting valid nodes from statdb %s", err)
+	}
+
+	var isValidNode bool
+	for i, node := range nodeIDs {
```

Comment:
Nested for loops are really inefficient. You could do this another way by doing something like this:
```
validNodes := make(map[storj.NodeID]bool)
for _, validID := range resp.PassedIds {
    validNodes[validID] = true
}
for i, nID := range nodeIDs {
    if !validNodes[nID] {
        invalidNodes = append(invalidNodes, int32(i))
    }
}
```
If you have 100 node IDs, a nested for loop means you need 100*100 iterations for the same logic that you can do in 200 iterations by using a map like above.

Reply:
done